pkg/devfile/adapters/common/command.go

```diff
@@ -7,140 +7,153 @@ import (
 	"github.com/openshift/odo/pkg/devfile/parser/data"
 	"github.com/openshift/odo/pkg/devfile/parser/data/common"
 	"k8s.io/klog"
-
-	"github.com/pkg/errors"
 )
 
 // GetCommand iterates through the devfile commands and returns the associated devfile command
-func getCommand(data data.DevfileData, commandName string, required bool) (supportedCommand common.DevfileCommand, err error) {
+func getCommand(data data.DevfileData, commandName string, groupType common.DevfileCommandGroupType) (supportedCommand common.DevfileCommand, err error) {
+
 	for _, command := range data.GetCommands() {
-		if command.Name == commandName {
 
-			// Get the supported actions
-			supportedCommandActions, err := getSupportedCommandActions(data, command)
+		command = updateGroupforCustomCommand(commandName, groupType, command)
```

Comment:
Actually, group for custom commands has to be inserted before `validateCommand`, otherwise it would fail with `group not found` . So it cannot bet inside `if commandName != ""`

Reply:
is the custom command update func call here for v1?

Reply:
@maysunfaisal we are updating group for commands specified by odo flag `--build-command`, `--run-command`, not for all commands. `updateGroupforCustomCommand`, we can remove the validation for group but group needs to be inserted for custom command, this is internally only so that user does not need to specify it in devfile.